CONTRIBUTING.md

```diff
@@ -2,6 +2,17 @@
 
 Contributions are welcome! 
 
+## Dependency Management
+
+The README states that we support `n-2` Argo CD versions, where `n` is the current minor version of Argo CD. 
+
+To do so we always import the exact Argo CD client-lib we provide support for.
```

Comment:
Not sure I follow. If we support `n-2` versions how can we import "the exact" client-lib we provide support for?

Reply:
I see that's a bit misleading. I meant if we support `2.14`, `2.13` and `2.12` we import the `2.14` client-lib. 

But this is a bit odd as well.

Reply:
Yeah, so historically we only bumped the client lib dependency when needed (security patches, new features being adopted) as this way we maintain compatability with older version for as long as possible.

Reply:
> But how then do you ensure it's working with the current versions of Argo CD that are supported upstream?

That is ensured by updating the versions of ArgoCD we test against in the test matrix in the GitHub action.

Reply:
All right, do you have a suggestion how to write this down in the contributor docs?